nodejs/eks/cluster.ts

```diff
@@ -184,38 +189,53 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
     // Create the VPC CNI management resource.
     const vpcCni = new VpcCni(`${name}-vpc-cni`, kubeconfig, args.vpcCniOptions, { parent: parent });
 
-    // Create the instance role we'll use for worker nodes.
-    let instanceRole: pulumi.Output<aws.iam.Role>;
-    if (args.instanceRole) {
-        instanceRole = pulumi.output(args.instanceRole);
+    const instanceRoleMappings: RoleMapping[] = [];
+    let instanceProfile: aws.iam.InstanceProfile;
+    if (args.instanceRoleArns && args.instanceRoleArns.length) {
+        // Create role mappings if an array of instanceRoleArn's is specified.
+        for (const [index, arn] of args.instanceRoleArns.entries()) {
+            const roleName = pulumi.output(arn).apply(r => r.split("/")).apply(r => r[1]);
+            const role = aws.iam.Role.get(`${name}-existingRole-${index}`, roleName);
```

Comment:
I don't think it's a good idea to use `.get`.  I had assumed we would just ask the user to pass us `Role`s, and they can do the `.get` if that's really what the need.  But more often they'll just construct the `Role` themselves.